resources/completion.zsh

```diff
@@ -1,16 +1,32 @@
 #compdef f3d
 
+local commandlist
 local longopts
 local shortopts
 local arguments
-local f3dhelp
 
-f3dhelp=$(f3d --help 2>&1 | sed '1,/Examples/!d')
+# commandlist operations:
+# Discard:
+#   - Header.
+#   - Keys/examples sections.
+#   - Section headers.
+#   - Illegal characters [].
+#   - Leading whitespace.
+# Join multiline helptext. See https://stackoverflow.com/a/37951863/4672189 for details.
+# Normalize whitespace.
+# Discard trailing whitespace.
+commandlist=$(
+  f3d --help 2>&1 |
+    tail -n +5 |
+    sed '1,/Keys:/!d' | sed '/.*:$/d' | sed 's/\[.*\] *//g' | sed 's/^\s*//' |
+    sed -e ':a' -e 'N' -e '/\n-[a-z\-]/!s/\n/ /' -e 'ta' -e 'P' -e 'D' | 
```

Comment:
Unfortunately, this line does not seem to work on macOS and remove everything.

Reply:
More specifically, this part causes issues: `/\n-[a-z\-]/!s/\n/ /`

Reply:
:slightly_frowning_face:

It looks like MacOS doesn't respect the `\n` escape. I tried using the `$'\n'` technique mentioned in that StackOverflow answer, but that breaks functionality in Linux.

I've just pushed 25e617f where I move that complex regex into a separate variable with a multiline string. This works on Linux, and the StackOverflow post suggests it should also work on MacOS.

Reply:
I'll check again

Reply:
Still failing:

```
➜  build git:(fix-zsh-completion) f3d --
sed: 1: "/\
-[a-z\-]/!s/\
/ /;
": unterminated regular expression

_arguments:comparguments:327: invalid argument: 
sed: 1: "/\
-[a-z\-]/!s/\
/ /;
": unterminated regular expression
_arguments:comparguments:327: invalid argument: 
sed: 1: "/\
-[a-z\-]/!s/\
/ /;
": unterminated regular expression
_arguments:comparguments:327: invalid argument: 
```

Reply:
I agree. Please remove the last two commits and it's good to go.

Reply:
:+1: I've moved those two commits to https://github.com/allemangD/f3d/tree/multiline-helptext. This PR should be good now.

Reply:
There is no problem with MacOS on master, it's just that the last improvements made by @allemangD to the autocomplete didn't work so he reverted them.

Reply:
ok, thats fair.